modules/platform/nuxeo-preview-core/src/main/java/org/nuxeo/ecm/platform/preview/adapter/ZipPreviewer.java

```diff
@@ -0,0 +1,131 @@
+import org.nuxeo.ecm.core.convert.api.ConversionService;
+import org.nuxeo.ecm.platform.htmlsanitizer.HtmlSanitizerService;
+import org.nuxeo.ecm.platform.mimetype.interfaces.MimetypeRegistry;
+import org.nuxeo.ecm.platform.preview.api.PreviewException;
+import org.nuxeo.ecm.platform.preview.helper.PreviewHelper;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Previewer for Zip blob.
+ * <p>
+ * It sanitizes each zip entry if needed.
+ *
+ * @since 11.5
+ */
+public class ZipPreviewer implements MimeTypePreviewer {
+
+    protected static final Set<String> HTML_MIME_TYPES = Set.of("text/html", "text/xml", "text/plain");
+
+    private static final Logger log = LogManager.getLogger(ZipPreviewer.class);
+
+    @Override
+    public List<Blob> getPreview(Blob blob, DocumentModel dm) throws PreviewException {
+        ConversionService conversionService = Framework.getService(ConversionService.class);
+        String converterName = conversionService.getConverterName("application/zip", "text/html");
+        if (converterName == null) {
+            throw new PreviewException("Unable to find zip2html converter");
+        }
+
+        try {
+            BlobHolder result = conversionService.convert(converterName, new SimpleBlobHolder(blob), null);
+            List<Blob> blobs = new ArrayList<>(result.getBlobs());
+            blobs.replaceAll(this::sanitize);
+            return blobs;
+        } catch (ConversionException e) {
+            throw new PreviewException(e.getMessage(), e);
+        }
+    }
+
+    protected Blob sanitize(Blob blob) throws PreviewException {
+        String filename = blob.getFilename();
+        if (!isSanitizable(blob)) {
+            log.debug("ZipEntryBlob: {} is not sanitizable", filename);
+            return blob;
+        }
+        try {
+            String content = blob.getString();
+            content = getHtmlSanitizerService().sanitizeString(content, null);
+            content = PreviewHelper.makeHtmlPage(content);
+            return Blobs.createBlob(content, "text/html", "UTF-8", filename);
+        } catch (IOException e) {
+            throw new PreviewException("Cannot fetch ZipEntryBlob content with filename: " + filename, e);
+        }
+    }
+
+    protected boolean isSanitizable(Blob blob) {
+        String mimeType = getMimetypeRegistry().getMimetypeFromFilenameWithBlobMimetypeFallback(blob.getFilename(), blob, null);
+        boolean sanitized = false;
+        if (mimeType == null) {
+            try (InputStream stream = blob.getStream()) {
+                // last chance introspect the content
+                LineIterator lineIt = IOUtils.lineIterator(stream, defaultIfNull(blob.getEncoding(), "UTF-8"));
+                while (lineIt.hasNext()) {
+                    String line = lineIt.nextLine();
+                    sanitized = line.contains("<script");
+                    if (sanitized) {
+                        break;
+                    }
```

Comment:
You could return directly here, no need for a local `sanitized` variable (which I'd have called `sanitizable`)